packages/marshal/src/rankOrder.js

```diff
@@ -99,7 +99,7 @@ const { entries, fromEntries, setPrototypeOf, is } = Object;
 const sameValueZero = (x, y) => x === y || is(x, y);
 
 const trivialComparator = (left, right) =>
-  // eslint-disable-next-line no-nested-ternary
+  // eslint-disable-next-line no-nested-ternary, @endo/restrict-comparison-operands
```

Comment:
wouldn't it be good to type the function signature so there are type errors if uncomparable are passed in? could be worth naming the type `number | bigint | string`.

Reply:
Do you know how to do that? The following is not enough:
```js
/**
 * @template {number | bigint | string} T
 * @param {T} left
 * @param {T} right 
 * @returns {RankComparison}
 */ 
const trivialComparator = (left, right) =>
  // eslint-disable-next-line no-nested-ternary
  left < right ? -1 : left === right ? 0 : 1;
```

Reply:
That should work, yeah. I suspect it's a limitation of how `restrict-comparison-operands` is implemented. Maybe it doesn't handle subtypes of the comparable types? `T extends string | number | bigint`

The suppression is fine in that case, but the limitation is worth documenting.